### PR TITLE
Added getAllEmployee method, refactor test cases for EmployeeTestCont…

### PR DESCRIPTION
**Pull request by rdmoraes**
Added new methods to return list of employees sorted by employee id. 
 

-  Get list of all employees
-  Get list of all managers
-  Get list of all staffs
- Re-wrote test cases in EmployeeControllerTest
- Added an interface in Employee class "Comparable<>" in order to sort List of employees

### CODE REVIEW

File: src/main/java/employee/com/Employee.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import java.util.Calendar;
 import java.util.Date;
 
-public abstract class Employee  {
+public abstract class Employee implements Comparable<Employee>  {
```
Comment (rdmoraes) [inline src/main/java/employee/com/Employee.java:11]: I needed to implement this interface in order to sort all results in the DataStorage.

File: src/main/java/employee/com/Employee.java [hunk 2]
```diff
@@ -141,6 +141,12 @@ public void checkPPSNoValidity(String ppsNo){
         }
     }
 
+    // override equals and hashCode
+    @Override
+    public int compareTo(Employee employee) {
+        return (Integer.parseInt(this.employeeId) - Integer.parseInt(employee.getEmployeeId()));
+    }
+
```
Comment (rdmoraes) [inline src/main/java/employee/com/Employee.java:149]: This method belongs to interface Comparable, and is implicitly used in the Collections.sort();

File: src/main/java/employee/com/EmployeeController.java [hunk 3]
```diff
@@ -39,6 +43,24 @@ public Employee getEmployeeDetails(String employeeId, boolean isManager) {
...
+
+    //Get All Employee Details
+    public List<Employee> getAllEmployeeDetails(){
+        List<Employee> totalEmployees = new ArrayList<>(staffService.getAllEmployeesDetails());
+        totalEmployees.addAll(managerService.getAllEmployeesDetails());
+        Collections.sort(totalEmployees);
+
+        return totalEmployees;
+    }
+
```
Comment (rdmoraes) [inline src/main/java/employee/com/EmployeeController.java:63]: I have here two methods with the same name.

-  The first method returns a list of Staffs or Managers isolated;
- The second method joins Staff and Managers in one single list and returns it a sorted by employee id

**Review by daniel-fitzg — APPROVED** [review]
LGTM

**Pull request merged**